dev/src/lobster/tools.h

```diff
@@ -1031,30 +1031,51 @@ template<typename T> T ReadMem(const void *p) {
     return dest;
 }
 
-template<typename T> T ReadMemInc(const uint8_t *&p) {
-    T dest = ReadMem<T>(p);
-    p += sizeof(T);
-    return dest;
+template<typename T> void WriteMemInc(uint8_t *&dest, const T &src) {
+    memcpy(dest, &src, sizeof(T));
+    dest += sizeof(T);
 }
 
-template<typename T, typename K = uint64_t> void ReadVec(const uint8_t *&p, T &v) {
-    auto len = ReadMemInc<K>(p);
-    v.resize((size_t)len);
-    auto blen = sizeof(typename T::value_type) * v.size();
-    memcpy(v.data(), p, blen);
-    p += blen;
+template<typename T> bool ReadSpan(const span<const uint8_t> &&p, T &v) {
```

Comment:
why the `&&` ?
Spans should probably be passed by value when possible.

Reply:
Ah sorry the c++11 stuff from long ago is leaking, with the const && it'll pass in the argument as an rvalue when it is a temporary.
Not relevant with a simple type like span here. i'll remove it